saddle-core/src/main/scala/org/saddle/scalar/ScalarTag.scala

```diff
@@ -22,18 +22,18 @@ import org.saddle.array.Sorter
 import org.saddle.Buffer
 import scala.collection.compat._
 import immutable.ArraySeq
+import org.saddle.array
 
 /** Typeclass definition for scalar tags. A ScalarTag contains important
   * meta-data regarding a scalar type, including how to instantiate a
   * Buffer/Vec/Mat/Index of that type, as well as an array. Often implicitly
   * required when dealing with objects in Saddle
   */
-trait ScalarTag[@spec(Boolean, Int, Long, Float, Double) T]
+trait ScalarTag[@spec T]
```

Comment:
The whole library specializes most often only Bool, Int, Long and Double. What is the purpose of this? Shouldn't saddle consistently specialize the same set or primitives everywhere?

Reply:
I see no reason why not specialize on char byte short etc as well. If you browse around the list in @spec(..) is fairly inconsistent. E.g. Float is missing sometimes.

Reply:
It was maybe done due to binary size? specialization can bloat it quickly and the original authors did consider that Bool, Int, Long and Double were used most often.